source/Halibut/HalibutRuntimeBuilder.cs

```diff
@@ -79,7 +79,9 @@ public HalibutRuntime Build()
             var serviceFactory = this.serviceFactory ?? new NullServiceFactory();
             if (serverCertificate == null) throw new ArgumentException($"Set a server certificate with {nameof(WithServerCertificate)} before calling {nameof(Build)}", nameof(serverCertificate));
             var logFactory = this.logFactory ?? new LogFactory();
-            var queueFactory = this.queueFactory ?? new DefaultPendingRequestQueueFactory(logFactory);
+#pragma warning disable CS0612
+            var queueFactory = this.queueFactory ?? (asyncHalibutFeature.IsEnabled() ? new PendingRequestQueueFactoryAsync(logFactory) : new DefaultPendingRequestQueueFactory(logFactory));
```

Comment:
If we want an async HalibutRuntime and have not specified an `IPendingRequestQueue` then get the async one by default